lib/buy/dfx/dfx_buy_provider.dart

```diff
@@ -159,6 +165,20 @@ class DFXBuyProvider extends BuyProvider {
 
   @override
   Future<void> launchProvider(BuildContext context, bool? isBuyAction) async {
+    if (wallet.isHardwareWallet) {
+      if (!getIt.get<LedgerViewModel>().isConnected) {
```

Comment:
**Minor:**
pass the LedgerViewModel as a dependency better

same for other providers